pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -890,8 +896,10 @@ func (s *DiscoveryServer) startPush(version string, push *model.PushContext, ful
 				version:            version,
 				edsUpdatedServices: edsOnly,
 			}:
+				client.mu.Lock()
```

Comment:
I understand the better safe than sorry approach (having a Lock() here + around the "added" part in StreamAggregatedResources().
If I understand correctly though, the adsClients should have unique TCP connection in it, and that race was created because of the "con.added" not being protected enough.
If we're seeing any lock contention, we should remopve the client.mu.Lock from this startPush() function, as it should be safe now that we made sure connections can not appear twice in the pending/adsClient arrays?

Reply:
Good catch, I tested without these and see no race still. I removed them